base/iterators.jl

```diff
@@ -1413,6 +1413,42 @@ Stacktrace:
     return ret
 end
 
+"""
+    getonly(f, x)
+
+Return the one and only element of collection `x`, or return `f()` if the collection has zero
+or multiple elements. `f()` will not be called if the collection has exactly one element.
+
+See also [`first`](@ref), [`last`](@ref), [`only`](@ref).
+
+!!! compat "Julia 1.9"
+    This method requires at least Julia 1.9.
+
+# Examples
+```jldoctest
+julia> getonly(('a', 'b')) do
+           'c'
+       end
+'c'
+
+julia> getonly(('a', 'b')) do
+           throw(ErrorException("My custom error message"))
+       end
+ERROR: My custom error message
+```
+"""
+@propagate_inbounds function getonly(f::F, x) where {F}
+    i = iterate(x)
+    @boundscheck if i === nothing
+        return f()
+    end
+    (ret, state) = i::NTuple{2,Any}
+    @boundscheck if iterate(x, state) !== nothing
+        return f()
+    end
```

Comment:
Those definitely shouldn't be annotated `@boundscheck`. `@inbounds only(f, x)` shouldn't imply that `x` contains exactly one element.

Reply:
Should we remove those annotations from the original `only(x)` method as well?

Reply:
I think those are different. There the alternative is always an error, so this can be thought of a kind of boundscheck. OTOH, you wouldn't use the two-arg form unless there were cases where `x` actually didn't contain exactly one element.

Reply:
I've removed the `@boundscheck` annotations from the `getonly(f, x)` method. I left the `only(x)` method unmodified.